lib/bin/run_test.js

```diff
@@ -255,9 +272,14 @@ function runTest(foundTest, runAfter) {
 
 process.on('uncaughtException', function(error) {
   sendError(error, { in: 'uncaught' });
-  exit(1);
+  cleanup(1);
+});
+
+process.on('beforeExit', function() {
```

Comment:
I was actually thinking of failing here since its not the normal path is it? normally we exit?